app/services/cloud_topology_service.rb

```diff
@@ -0,0 +1,74 @@
+class CloudTopologyService < TopologyService
+  include UiServiceMixin
+
+  @provider_class = ManageIQ::Providers::CloudManager
+
+  def entity_type(entity)
+    entity.class.name.demodulize
+  end
+
+  def build_topology
+    topo_items = {}
+    links = []
+
+    included_relations = [
```

Comment:
I would extend and change the relations like this:
  
     :tags,
     :availability_zones => [
       :tags,
       :vms => [
         :tags,
         :host,
         :service,
         :genealogy_parent, # image
         :flavor],
     :cloud_tenants      => [
       :tags,
       :children, # cloud tenants can be a tree
       :vms],
     :orchestration_stacks => [
        :tags,
        :vms,
        :service,
        :children, # stacks can be a tree
        :cloud_tenant,
        :orchestration_template ]

optionally, put tags to all the relations

^ should be a sane set to show in cloud, some of them might go to other pages later